lib/inject.rb

```diff
@@ -37,13 +55,18 @@ class Category < ActiveRecord::Base
   #
   # For example, the query "hey there" should match any category with a body
   # containing both "hey" and "there."
-  def self.search(query)
+  #def self.search(query)
     # TODO: rewrite to use inject instead of each and mutation
-    relation = self
-    query.split(' ').each do |keyword|
-      relation = relation.where('body LIKE ?', "%#{keyword}%")
+    #relation = self
+    #query.split(' ').each do |keyword|
+      #relation = relation.where('body LIKE ?', "%#{keyword}%")
+    #end
+    #relation
+
+  def self.search(query)
+    query.split(' ').inject(self.all) do |entries, search_term|
+      entries.where('body LIKE ?', "%#{search_term}%")
```

Comment:
This seems a little bit convoluted. That said, it's reasonably efficient. Maybe just moving the `query.split` onto its own line would be good.